iseq.c

```diff
@@ -1391,6 +1391,47 @@ iseqw_s_compile(int argc, VALUE *argv, VALUE self)
     return iseqw_new(rb_iseq_compile_with_option(src, file, path, line, opt));
 }
 
+static VALUE
+iseqw_s_compile_prism_compile(pm_parser_t *parser, VALUE opt, rb_iseq_t *iseq, VALUE file, VALUE path, int first_lineno) {
+    pm_node_t *node = pm_parse(parser);
+    pm_line_column_t start_line_column = pm_newline_list_line_column(&parser->newline_list, node->location.start);
+    pm_line_column_t end_line_column = pm_newline_list_line_column(&parser->newline_list, node->location.end);
+
+    rb_compile_option_t option;
+    make_compile_option(&option, opt);
+
+    prepare_iseq_build(
+        iseq,
+        rb_fstring_lit("<compiled>"),
+        file,
+        path,
+        first_lineno,
+        &(rb_code_location_t) {
```

Comment:
Should we extract this and the `start/end_line_column` to its own function that takes a `pm_newline_list_t` and a `pm_location_t` since we're reusing it